themes/theme-gmd/pages/Product/components/Header/components/CTAButtons/components/CartButton/index.jsx

```diff
@@ -80,26 +85,43 @@ class CartButton extends Component {
    * all criteria set by the conditioner are met.
    */
   handleClick = () => {
+    const {
+      disabled, conditioner, addToCart, productId, options, userLocation,
+    } = this.props;
+
     if (this.state.clicked) {
       return;
     }
 
-    if (this.props.disabled) {
+    if (disabled) {
       return;
     }
 
-    this.props.conditioner.check().then((fulfilled) => {
+    conditioner.check().then((fulfilled) => {
       if (!fulfilled) {
         return;
       }
 
       this.setState({ clicked: true });
 
-      this.props.addToCart({
-        productId: this.props.productId,
-        options: this.props.options,
+      const addToCartData = {
+        productId,
+        options,
         quantity: this.context.quantity,
-      });
+      };
+
+      // Add the user location for ROPIS if it is set.
+      if (userLocation !== null) {
```

Comment:
Now the add to cart request will always be dispatched with the fulfillment method after the user has selected a location once. It doesn't matter if the user has "Direct ship" enabled, or the product has fulfillment methods.

Same for iOS.